permuta/perm_utils/polynomial.py

```diff
@@ -0,0 +1,109 @@
+from collections import deque
+from itertools import islice
+from typing import ClassVar, Deque, Dict, FrozenSet, Iterable, Iterator, Tuple
+
+from permuta.patterns.perm import Perm
+
+
+class PolyPerms:
+    """A static container of methods to check if a perm set has a polynomial growth."""
+
+    PERM_TYPE_0: ClassVar[int] = 0  # W++
```

Comment:
Would it make sense to use an Enum to group those constant together?

Reply:
I tried 
```py
a,b,c = range(3)
```
but couldn't get typing to work for it. You have any suggestions?

Reply:
I thought about using that [the Enum class](https://docs.python.org/3/library/enum.html) to group the type into a meaningful unit

Reply:
```py
class PermType(IntEnum):
    """Collection of structural types for perms"""

    WPP: ClassVar[int] = 0  # W++
    WPM: ClassVar[int] = 1  # W+-
    WMP: ClassVar[int] = 2  # W-+
    WMM: ClassVar[int] = 3  # W--
    WIPP: ClassVar[int] = 4  # Winv++
    WIPM: ClassVar[int] = 5  # Winv+-
    WIMP: ClassVar[int] = 6  # Winv-+
    WIMM: ClassVar[int] = 7  # Winv--
    L2: ClassVar[int] = 8  # L2
    L2I: ClassVar[int] = 9  # L2inv
```

Reply:
Alternative:
```py
class PermType(IntEnum):
    """Collection of structural types for perms"""

    WPP, WPM, WMP, WMM, WIPP, WIPM, WIMP, WIMM, L2, L2I = range(10)
```

Reply:
I like the verbosity of the first one. I also think you want plain Enum not IntEnum since your type are not comparable

Reply:
Also I think you can remove the type annotation for the class var